webapp/app/model/recursive_data.py

```diff
@@ -24,24 +42,41 @@ def _update_input(self, data: Any) -> Any:
         fields = list(self.__class__.__fields__.values())
         self.__class__._previous_fields = []
 
+        non_found = False
         for field in fields:
             if issubclass(field.type_, BaseModel):
                 self.__class__._previous_fields.append(field.name)
-                self._set_data_for_previous_field(enriched_data, field.name, field.type_)
+                if self._set_data_for_previous_field(enriched_data, field.name, field.type_):
+                   non_found = True
 
+        if non_found:
+            print(self.__class__.__name__)
```

Comment:
This was only added for testing purposes. That said, we have a slight problem with the recursive data model, as it might be harder to understand why something failed in the upper chain